WordPress/gutenberg
# Remove compatibility layer for WP 6.2

## What?


Gutenberg only supports the current and previous WP version, so now that 6.4 is out, the compatibility files supporting 6.2 can be removed. This PR removes those files and fixes any obvious resulting fatals, but I'd thank everyone, especially those who have made PHP changes recently, to test that stuff is still working! 

For reference, the corresponding PR after 6.3: #53451

## Testing Instructions





Open all the pages, click around, check nothing's broken. In particular, there's a lot of changes in here related to templates, patterns, navigation and revisions so those areas are good to pay special attention to.
